src/main/java/net/minecraftforge/fmp/item/PartPlacementWrapper.java

```diff
@@ -0,0 +1,212 @@
+package net.minecraftforge.fmp.item;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.common.base.Predicate;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.SoundType;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemBlock;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.entity.player.PlayerInteractEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fmp.multipart.IMultipart;
+import net.minecraftforge.fmp.multipart.MultipartHelper;
+import net.minecraftforge.fmp.network.MessageWrappedPartPlacement;
+import net.minecraftforge.fmp.raytrace.RayTraceUtils;
+
+public class PartPlacementWrapper
```

Comment:
It handles the placement of things like multiparts in the case that they are using an ItemBlock, for example. So if by default it's a block (which can be converted to a part and back), if you attempt to place it inside a block that already has multiparts it'll place a part instead. There's also an adapted version for microblock containers so that they can be placed if there are already microblocks in that blockspace.